### DISCUSSION TITLE
Server lagging and items dupping

### DISCUSSION THREAD

**Discussion by KrazyDiamond** [Pryaxis/TShock#2232] [Q&A]
Hey all, need some help here...

I am running a tshock server for a handful of friends and family.
Most online at a time is only 4 players.
The server is running on an RPi 4B with 4GB of RAM using Mono.
Gigabit Down / 100Mbps Up (600/30 typical)

The server often experiences latency issues (lag, drops disappearing and reappearing, enemies teleporting, time jumping backwards), which seems to have gotten noticeably worse as the server has progressed. I am assuming that the RPi has a difficult time keeping up because the server is running through Mono and that is resource-intensive.

Events with a lot of enemies are one of the worst. By far the clentaminator being used by anyone is a complete mess, nearly locking up the game for anyone on the server. When the server lags hard during drops, such as the killing of a boss, the items often drop, are picked up by someone, then the server corrects itself and the items drop/spawn again from the kill point and if anyone had already grabbed them falling then the items duplicate.

Are there solutions to this or something that can be done to improve it? Either to prevent the server from lagging so hard, duplicating items, or to make an RPi4 run tshock smoother?

**Comment by KrazyDiamond** [comment]
No one else experience this, or have any recommendations?

**Comment by rusocop** [comment]
The internet is ok and PC's statistics too. Try to restart it everyday in case you have it online 24/7, maybe the Beta version of TShock is "causing" the problem. Re-view all your custom permissions or parameters if you have some of those created/changed. My friend has 200MB up and down internet, Windows 10, i9-10900k, 32 GB RAM even GTX 3090 and we still experience some of your kind of lag too...

**Comment by Thunder1723** [comment]
I also have a pi 4 but I got it in the 8gb version, doesn't mean anything for running a Tshock server though, I would try connecting your pi to a Ethernet port.  If that doesn't work I would try overclocking the pi.  I have mine running full blast when players join and I idle at 1Ghz.  It may seem a bit scary running the pi to its limits but as long as you get a good cooling solution it should be fine.  There are thousands of guides on how to overclock the pi 4.  As for a cooling solution I use this cooler https://www.amazon.com/gp/product/B07ZV1LLWK/ref=ppx_yo_dt_b_asin_title_o01_s00?ie=UTF8&psc=1 in combination with kryonaut thermal paste.  My idle temp is 30 C and my highest achieved temp so far is 39C.  You may also run into the problem where the pi won't boost past 1.9Ghz and there is nothing you can do about it (literally, look up "silicon lottery").  I also got a Noctua fan because I thought that cooler didn't come with a non RGB fan.  It does.  So if you don't want to drop a extra $14 on a fan and you don't want RGB, you don't have to.
